flink-connectors/flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSAuthenticationException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.aws.util;
+
+import org.apache.flink.annotation.Internal;
+
+/**
+ * A {@link IllegalStateException} wrapper indicating the exception was thrown from AWS credentials.
+ */
+@Internal
+public class AWSAuthenticationException extends IllegalStateException {
```

Comment:
I do not think `IllegalStateException` semantically fits here:

> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.

https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalStateException.html

We are throwing this when bad creds are provided right? So a better fit might be `IllegalArgumentException`? In any case, the purpose of this class is not really clear. If we are just wrapping any and all exceptions that arise from authentication then we should extend `RuntimeException`